dev/com.ibm.ws.ejbcontainer.remote/src/com/ibm/ws/ejbcontainer/remote/internal/EJBRemoteRuntimeImpl.java

```diff
@@ -249,17 +250,25 @@ public synchronized void unbindAll(Object bindingDataObject) {
             }
         }
 
-        for (int i = bindingData.contextNames.length - 1; i >= 0 && !hasBindings(bindingData.contexts[i]); i--) {
-            NamingContext parentContext = i == 0 ? rootNamingContext : bindingData.contexts[i - 1];
-            unbind(parentContext, bindingData.contextNames[i]);
+        // For contextNames, if completely empty of bindings, unbind the naming context from its parent context.
+        for (int i = bindingData.contextNames.length - 1; i >= 0; i--) {
+            if (isBindingContextEmpty(bindingData.contexts[i])) {
```

Comment:
This is better, but inefficient.  The code used to stop at the first non-empty list, but will now keep going.  The logic here is skipped, but `isBindingContextEmpty` will continue to be called and creates several objects and creates a list.  Could we have `isBindingContextEmpty` return `Boolean`, so we could tell the difference between True, False, and null?